libslide/src/parser.rs

```diff
@@ -0,0 +1,63 @@
+use crate::scanner::types::Token;
+
+pub enum Expr {
+    Float(f64),
+    Int(i64),
+    BinOp(Box<BinOp>),
+    // I added un op even though I havent implemented + or -
+    UnaryOp(Box<UnaryOp>),
+}
+
+pub struct BinOp {
+    pub item: Token,
+    pub lhs: Expr,
+    pub rhs: Expr
```

Comment:
It should be enough to make these `Box<Expr>`. Then you will be able to construct a `BinOp` with any expression.